src/components/base/Dialogs/LoadingDialog.tsx

```diff
@@ -0,0 +1,56 @@
+import { RepeatIcon } from "@chakra-ui/icons";
+import { Button, Modal, ModalBody, ModalContent, ModalOverlay, Text } from "@chakra-ui/react";
+
+interface ButtonConfig {
+  text: string;
+  onClick: () => void;
+}
+interface LoadingDialogProps {
+  message: string;
+  isOpen: boolean;
+  onClose?: () => void;
+  width?: number;
+  icon?: React.ReactNode;
+  buttonConfig?: ButtonConfig;
+}
+
+const LoadingDialog = (props: LoadingDialogProps) => {
+  const { message, isOpen, onClose, width, icon, buttonConfig } = props;
+  return (
+    <>
+      <Modal
+        isOpen={isOpen}
+        isCentered={true}
+        closeOnOverlayClick={false}
+        closeOnEsc={false}
+        onClose={onClose ? onClose : () => null}
+      >
+        <ModalOverlay />
+        <ModalContent width={width ? `${width}px` : "317px"}>
+          <ModalBody
+            width={"100%"}
```

Comment:
The inline styling is ok for the short term but we should start using the Chakra Styling API to manage styles in a separate file.